packages/stateManager/tests/stateManager.test.ts

```diff
@@ -0,0 +1,343 @@
+  });
+  const vectorState = testVectors.reduced;
+
+  test("ShopManifest - adds and updates shop manifest events", async () => {
+    const shop = await stateManager.manifest.get();
+    expect(shop.name).toEqual(vectorState.manifest.name);
+    expect(shop.profilePictureUrl).toEqual(
+      vectorState.manifest.profile_picture_url,
+    );
+    expect(shop.description).toEqual(vectorState.manifest.description);
+    expect(shop.publishedTagId).toEqual(vectorState.manifest.published_tag);
+    expect(shop.addAcceptedCurrencies!.length).toEqual(
+      vectorState.manifest.accepted_currencies.length,
+    );
+  });
+
+  test("ListingManager - adds and updates item events", async () => {
+    const listingIterator = stateManager.items.iterator;
+    for await (const [id, item] of listingIterator) {
+      const vectorItem = vectorState.items[id];
+      expect(item.price).toEqual(vectorItem.price);
+      const parsed = JSON.parse(vectorItem.metadata);
+      expect(item.metadata.title).toEqual(parsed.title);
+      expect(item.metadata.image).toEqual(parsed.image);
+      expect(item.quantity).toEqual(vectorItem.stock_qty);
+    }
+  });
+
+  test("TagManager - adds and updates tag events", async () => {
+    const tagIterator = stateManager.tags.iterator;
+    const { tags } = vectorState;
+    for await (const [id, tag] of tagIterator) {
+      expect(tags[id].name).toEqual(tag.name);
+    }
+  });
+
+  test("OrderManager - adds and updates order events", async () => {
+    const { payed, abandoned, committed, open } = vectorState.orders;
+    //If order has a txHash that means the order has been payed and status should be marked as complete.
+    //So we are checking that all payed orders have status=complete and that it has a txHash
+    for await (const order of payed) {
+      const o = await stateManager.orders.get(order.order_id as `0x${string}`);
+      expect(o.status).toEqual("COMPLETE");
+      expect(o.txHash).toEqual(order.tx_hash);
+    }
+    //Abandoned orders should be stauts=failed
+    for await (const order of abandoned) {
+      const o = await stateManager.orders.get(order.order_id as `0x${string}`);
+      expect(o.status).toEqual("FAILED");
+    }
+    //Commited orders should have the items + qty purchased, and the total amount.
+    for await (const order of committed) {
+      const o = await stateManager.orders.get(order.order_id as `0x${string}`);
+      expect(o.items).toEqual(order.items);
+    }
+    //Open orders should have status=pending and have the corrects items + qty
+    for await (const order of open) {
+      const o = await stateManager.orders.get(order.order_id as `0x${string}`);
+      expect(o.status).toEqual("PENDING");
+      expect(o.items).toEqual(order.items);
+    }
+  });
+});
+
+describe("CRUD functions update stores", async () => {
+  const client = new MockClient();
+  const stateManager = new StateManager(
+    // @ts-expect-error FIXME looking for client to be RelayClient
```

Comment:
we can redefine what the stateManager takes as client type to accommodate both MockClient and RelayClient. Especially since the state manager also does not utilise all the relay client functions